src/app/modules/main/chat_section/chat_content/controller.nim

```diff
@@ -53,132 +53,158 @@ proc newController*(delegate: io_interface.AccessInterface, events: EventEmitter
   result.communityService = communityService
   result.messageService = messageService
 
+proc disconnect*(self: Controller)=
+  for id in self.connectionIds:
+    self.events.disconnect(id)
+  self.connectionIds = @[]
+
 proc delete*(self: Controller) =
-  discard
+  self.disconnect()
 
 proc init*(self: Controller) =
-  self.events.on(SIGNAL_MESSAGES_LOADED) do(e:Args):
+  var handlerId = self.events.onWithUUID(SIGNAL_MESSAGES_LOADED) do(e:Args):
```

Comment:
This seems very repetitive and annoying to do. Maybe we could create a new function on `events` so that we pass it a UUID that represents the module (controller) that calls it, so then we just call the events to remove the connections for that module. Eg:

```nim
# Creationg of event handlers, almost the same as before, but we pass it our UUID
self.events.onWithModuleUUID(self.myModuleUUID, SIGNAL_MESSAGES_LOADED) do(e:Args):
  # Then normal event handling


# In the events class
self.moduleEvents = initTable[string,seq[string]]() # moduleUUID, seq[eventUUIDs]

# when onWithModuleUUID is called, we do what you did here, but add it to the table
var handlerId = self.events.onWithUUID(eventName) do(e:Args):
  proc()
self.moduleEvents[moduleUUID].add(handlerId) # with checks to make sure the seq was initiated

# When we delete the module, just call the disconnect function on events
self.events.disconnectModule(self.myModuleUUID)
# And the events controller then just loops the events for that uuid and clears the table entry.
```

Does it make any sense? :laughing: 

Anyway, it would simplify a lot and also would make it work for all controllers if we want.